### PR TITLE
PDEP6 implementation pt 2: eablock.setitem, eablock.putmask

### PR DESCRIPTION
**Pull request by MarcoGorelli**
following on from #50626

### CODE REVIEW

File: doc/source/whatsnew/v2.1.0.rst [hunk 1]
```diff
@@ -295,8 +295,86 @@ Other API changes
...
 
-Deprecate parsing datetimes with mixed time zones
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Deprecations
+~~~~~~~~~~~~
+
+Deprecated silent upcasting in setitem-like Series operations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Setitem-like operations on Series (or DataFrame columns) which silently upcast the dtype are
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v2.1.0.rst:304]: Worth listing what methods are setitem-like?

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @MarcoGorelli

**Comment by mroeschke** [comment]
@MarcoGorelli do you have time/appetite to start enforcing the PDEP6 deprecation?

If not, is enforcing the deprecation as easy as switching the warnings to exceptions or is there more cleanup?

**Comment by MarcoGorelli** [comment]
thanks for the ping! yup, I'll get to it - exciting to see it enforced!

**Comment by MarcoGorelli** [comment]
> @MarcoGorelli do you have time/appetite to start enforcing the PDEP6 deprecation?

doing it today - sorry for the delay, been too caught up with other work projects, but some of them are winding down now

**Pull request merged**